.github/workflows/job-precommit-sql.yml

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+# Runs Runs Java PreCommit SQL scheduled on cron, on push to master and on pull request
+
+name: PreCommit SQL
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/java/extensions/sql**']
+permissions: read-all
+
+jobs:
+  set-properties:
+    runs-on: [self-hosted, ubuntu-20.04]
+    outputs:
+      properties: ${{ steps.test-properties.outputs.properties }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+          submodules: false
+      - id: test-properties
+        uses: ./.github/actions/setup-default-test-properties
+
+  sql-precommit:
+    needs: set-properties
+    name: Run PreCommit SQL
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        exclude: #Skipping Java 11 and 17 due to issues caused by new access options added in JDK 9 and onwards.
+          - java: "11"
+          - java: "17"
+    timeout-minutes: 180
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+          submodules: false
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+      - name: Remove default github maven configuration
+        # This step is a workaround to avoid a decryption issue of Beam's
+        # gradle-command-action plugin and github's provided maven
+        # settings.xml file
+        shell: bash
+        run: rm ~/.m2/settings.xml
```

Comment:
Is this something we should include in the gradle-command-self-hosted-action itself rather than putting it in multiple workflows?

Reply:
It is included in the setup-self-hosted-action yaml, but for some reason we haven't determined yet the settings.xml file gets created again prior to this step so we have to remove it again.

Reply:
My 2 guesses are that either its caused by setup-java or that its because we're in a different job. Either way, I'd probably recommend we just add it to the `gradle-command-self-hosted-action`. If its already removed, then great, if its not then it won't hurt to remove it.